gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1483,7 +1491,41 @@ private Credential getCredential(Configuration config)
       }
     }
 
-    return credential;
+    // If impersonation service account exists, then use current credential to request access token
+    // for the impersonating service account.
+    return getImpersonationCredential(config, GCS_CONFIG_PREFIX, credential).orElse(credential);
+  }
+
+  /**
+   * Generate a {@link Credential} from the internal access token provider based on the service
+   * account to impersonate.
+   */
+  private static Optional<Credential> getImpersonationCredential(
+      Configuration config, String keyPrefix, Credential credential) throws IOException {
+    String impersonationServiceAccount =
+        HadoopCredentialConfiguration.getImpersonationServiceAccount(config, keyPrefix);
```

Comment:
Inline this method and use [`CONFIG_KEY_PREFIXES`](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/f122baabed8c9a90a8fada5d1486b39ebdcfcab9/gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java#L55)  as a prefixes.

Reply:
Same variable is referenced below, so better keep it. Prefix name updated.

Reply:
I meant `getImpersonationServiceAccount` method, we should inline because it has no logic and also we need to use [`CONFIG_KEY_PREFIXES`](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/f122baabed8c9a90a8fada5d1486b39ebdcfcab9/gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java#L55) constant to get property value.

Reply:
Done.